src/EoS/Nuclear/NuclearEoS.cpp

```diff
@@ -185,6 +184,10 @@ void nuc_eos_C_short( real *Out, const real *In,
 #        endif
 
 #        if ( NUC_EOS_SOLVER != NUC_EOS_SOLVER_ORIG )
+         if ( leps >  table_chk[npt_chk-1]  )  {   in_aux_table = false;   }
+         if ( leps <  table_chk[        0]  )  {   in_aux_table = false;   }
+#        endif
+#        if ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
          if ( leps >  table_chk[npt_chk-1]  )  {  *keyerr = 110;  return;  }
          if ( leps <  table_chk[        0]  )  {  *keyerr = 111;  return;  }
 #        endif
```

Comment:
1. Rewrite these codes. In your implementation, we do the check twice if `NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY` and `NUC_EOS_SOLVER != NUC_EOS_SOLVER_ORIG`. 

2. Also, I think the original codes are also problematic.

3. For consistency, use `NUC_TABLE_MODE != NUC_TABLE_MODE_ENGY` instead if you still need this condition.

Reply:
Change it to  NUC_TABLE_MODE != NUC_TABLE_MODE_ENGY

Reply:
1. On comments 3, my bad. I meant to use `NUC_TABLE_MODE_TEMP` for consistency.
2. In case of `NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP`, there is no auxiliary table if the original solver is adopted.

Reply:
I changed this part to `NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP`

Reply:
- We still need to check the energy even using the original solver.
- Add some comments to indicates which `if` macro the blocks are belong to.

Reply:
- I don't get it. We are checking `if ( leps != leps                  )`. And in the case of `ORIG` we don't have auxiliary table for energy.

- I added comments for that.

Reply:
- In the `ORIG` case, the table could be based on energy, and thus we need to check if the input energy is valid.
- Minimize the comments, otherwise it make it difficult to read for me. (e.g., the one for `if...else` should be enough).

Reply:
I separated those and checked the case when `ORIG` and `ENGY`.

Reply:
Combine them together....

Reply:
I combined it.

Reply:
Combine them together without implementing the following code twice:
~~~ c
         if ( leps >  table_chk[npt_chk-1]  )  {  *keyerr = 110;  return;  }
         if ( leps <  table_chk[        0]  )  {  *keyerr = 111;  return;  }
~~~

If you still do not know how to do that, I will show you the solution later.

Reply:
I updated this.